tests/tensorflow2/test_grad_tape_tf_function.py

```diff
@@ -0,0 +1,81 @@
+# Third Party
+import tensorflow as tf
+
+# First Party
+import smdebug.tensorflow as smd
+from smdebug.core.collection import CollectionKeys
+
+
+def create_hook(trial_dir):
+    hook = smd.KerasHook(trial_dir, save_all=True)
+    return hook
+
+
+def create_model():
+    model = tf.keras.models.Sequential(
+        [
+            # WA for TF issue https://github.com/tensorflow/tensorflow/issues/36279
+            tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
+            tf.keras.layers.Dense(128, activation="relu"),
+            tf.keras.layers.Dropout(0.2),
+            tf.keras.layers.Dense(10, activation="softmax"),
+        ]
+    )
+    return model
+
+
+def test_gradtape_tf_function(out_dir):
```

Comment:
do we have to update the quip doc or general docs about what is supported in this combination?

Reply:
Yes. Or we can add this example to the examples repository.